samples: subsys: llext: modules: sample.llext.modules.module_build reports overflow

**Describe the bug**
Sample `samples/subsys/llext/modules/sample.llext.modules.module_build` report overflow on RAM region.

Observed for:
- `nrf52dk/nrf52832`
- `nrf52840dk/nrf52840`
- `nrf5340dk/nrf5340/cpuapp`
- `nrf5340dk/nrf5340/cpuapp/ns`
- `nrf5340dk/nrf5340/cpunet`
- `nrf54l15pdk/nrf54l15/cpuapp`
- `nrf9160dk/nrf9160`
- `nrf9160dk/nrf9160/ns`

**To Reproduce**
Steps to reproduce the behavior:
1. have `nrf52840dk/nrf52840` connected
2. go to your zephyr dir
3. call `./scripts/twister -T samples/subsys/llext/modules -p nrf52840dk/nrf52840 --device-testing --device-serial /dev/ttyACM0 -v --inline-logs --west-flash="--erase"`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
*** Booting Zephyr OS build v3.7.0-rc1-80-g694428947ae3 ***
[00:00:00.254,760] <inf> app: Calling hello world as a module
[00:00:00.260,986] <err> os: ***** USAGE FAULT *****
[00:00:00.266,571] <err> os:   Stack overflow (context area not valid)
[00:00:00.273,742] <err> os: r0/a1:  0xb00bd09e  r1/a2:  0xacfbbeff  r2/a3:  0x00002256
[00:00:00.282,348] <err> os: r3/a4:  0x29000200 r12/ip:  0x200112c0 r14/lr:  0x00009068
[00:00:00.290,985] <err> os:  xpsr:  0x00000000
[00:00:00.296,173] <err> os: Faulting instruction address (r15/pc): 0x000003bb
[00:00:00.304,016] <err> os: >>> ZEPHYR FATAL ERROR 2: Stack overflow on CPU 0
[00:00:00.311,889] <err> os: Current thread: 0x20000658 (unknown)
[00:00:00.318,603] <err> os: Halting system
```

Sample do not build on `nrf52dk/nrf52832` and `nrf5340dk/nrf5340/cpunet`:
```
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `noinit' will not fit in region `RAM'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: region `RAM' overflowed by 8752 bytes
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain: Zephyr SDK 0.16.5-1
 - Commit SHA or Version used: `v3.7.0-rc1-24-g0561d74d31c4`

**Additional context**
Probably introduced by PR #74060

--- a/samples/subsys/llext/modules/prj.conf
+++ b/samples/subsys/llext/modules/prj.conf
@@ -12,3 +12,8 @@ CONFIG_LLEXT=y
 CONFIG_LLEXT_LOG_LEVEL_DBG=y
 CONFIG_LLEXT_HEAP_SIZE=64
 CONFIG_LLEXT_TYPE_ELF_RELOCATABLE=y
+
+# This test consumes large amounts of stack when loading the LLEXT.
+# Increase the available size to avoid overflowing (see issue #74536).
+
+CONFIG_MAIN_STACK_SIZE=2048

